src/stores/EmailFormStore.ts

```diff
@@ -0,0 +1,172 @@
+
+    // We assume that first domain is domain of truth and compare all others to it
+    const firstDomain = this.getDomain(firstEmail)
+
+    // Checks whole list when we remove email
+    if (!fileName || !email) {
+      this.hasDifferentDomains = false
+      const arrays = Object.values(this.emailMapping)
+
+      for (const emails of arrays)
+        for (const email of emails) {
+          if (this.getDomain(email.email) === firstDomain) continue
+
+          this.hasDifferentDomains = true
+          email.isOtherDomain = true
+        }
+
+      return
+    }
+
+    // Checks when we add email
+    const inputDomain = this.getDomain(email)
+
+    if (firstDomain === inputDomain) {
+      this.hasDifferentDomains = false
+      return
+    }
+
+    this.hasDifferentDomains = true
+  }
+
+  private getDomain(email: string) {
+    return email.split('@')[1]
+  }
+
+  private checkDuplicates(inputEmail: string, inFile?: boolean) {
+    if (!Object.keys(this.emailMapping)) return true
+
+    const hasDuplicates = Object.values(this.emailMapping)
+      .flat()
+      .filter(({ email }) => email === inputEmail)
+
+    if (!hasDuplicates.length) return true
+
+    if (!this.warnedAboutDuplicates)
+      toast.warn("Duplicate emails don't make you more anonymous 👀")
+    if (inFile) this.warnedAboutDuplicates = true
+    return false
+  }
+
+  private deleteFile(fileName: string) {
+    delete this.emailMapping[fileName]
+  }
+
+  private setAllIsNotSameDomain() {
+    for (const emails of Object.values(this.emailMapping)) {
+      for (const email of emails) {
+        email.isOtherDomain = false
+      }
+    }
+  }
+
+  private removeMappingIfEmpty(fileName: string) {
+    if (!this.emailMapping[fileName].length) this.deleteFile(fileName)
+  }
+
+  setEmailListFromFile(stringList: string, fileName: string) {
+    for (const emailArray of stringList.matchAll(fileEmailRegex)) {
+      const email = emailArray[0]
+      if (!this.checkDuplicates.bind(this)(email, true)) continue
+
+      this.checkSameDomain.bind(this)(fileName, email)
+      this.createOrPush(fileName, email, this.hasDifferentDomains)
+    }
+    this.warnedAboutDuplicates = false
+  }
+
+  removeLastEmail() {
+    if (this.inputEmail) return
+
+    const lastFilename = Array.from(Object.keys(this.emailMapping)).pop()
+    if (!lastFilename) return
+
+    this.emailMapping[lastFilename].pop()
+    this.removeMappingIfEmpty(lastFilename)
+    this.checkSameDomain()
+  }
+
+  removeEmailsFromList(fileName: string, index?: number) {
+    if (index === undefined) {
+      // We don't know, if the first email has changed, so we check all of them
+      this.setAllIsNotSameDomain()
+      this.deleteFile(fileName)
+    } else {
+      this.emailMapping[fileName] = this.emailMapping[fileName].filter(
+        (_, indexToCheck) => indexToCheck !== index
+      )
+      // If we remove first email, than we update the domain of truth
```

Comment:
If the use removes first email, we need to check if other emails has valid domain according to the new first email (and domain)